src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts

```diff
@@ -56,6 +64,40 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
       });
   }
 
+  private deleteFriendsFromList(id, array) {
+    const indexSuggestion = array.findIndex((item) => item.id === id);
+    array.splice(indexSuggestion, 1);
```

Comment:
the index here can be -1
I would return new array here:
return array.filter((item) => item.id !== id)